app/javascript/util/languageUtil.tsx

```diff
@@ -227,3 +231,31 @@ export function localizedFormat(date: string, format: string): string {
 
   return dayjs(date).locale(dayJsLocales[lang]).format(format)
 }
+
+/**
+ * Get the current language prefix as a capitalized string, or default to the english prefix if there is no explicit prefix in
+ * the path.
+ * @returns {string} - language code: ES, ZH, TL, EN
+ */
+export const getPageLanguageCode = (): string => {
+  const languageInRoute = getCurrentLanguage()
+  const languageConfig = LANGUAGE_CONFIGS[languageInRoute]
+  return RailsTranslationLanguage[languageConfig?.prefix.toUpperCase()]
+}
+
+/**
+ * Get the translated string for a given field name from the translations object.
+ * @param originalValue {string} - original value of the field
+ * @param fieldName {string} - field name to get the translation for in Saleforce notation: Credit_Rating__c
+ * @param translations {[key: string]: RailsTranslation} - translations object for the listing
+ * @returns {string} - translated string for current language or original value if no translation exists
+ */
+export const getTranslatedString = (
+  originalValue: string,
+  fieldName: string,
+  translations: RailsTranslations
+) => {
+  const languageCode = getPageLanguageCode()
+  const translatedValue = translations ? (translations[fieldName]?.[languageCode] as string) : null
```

Comment:
you need like an isEmpty(translations) instead of just translations, because translations is coming in as an empty object and that is true for the purposes of the ternary